app/serializers/export/own_post_serializer.rb

```diff
@@ -0,0 +1,33 @@
+module Export
+  # This is a serializer for the user's own posts
+  class OwnPostSerializer < FederationEntitySerializer
+    # Only for public posts.
+    # Includes URIs of pods which must be notified on the post updates.
+    # Must always include local pod URI since we will want all the updates on the post if user migrates.
+    has_many :subscribed_pods_uris
+
+    # Only for private posts.
+    # Includes diaspora* IDs of people who must be notified on post updates.
+    has_many :subscribed_users_ids
+
+    # Normally accepts Post as an object.
+    def initialize(*)
+      super
+      self.except = [excluded_subscription_key]
+    end
+
+    private
+
+    def subscribed_pods_uris
+      object.subscribed_pods_uris.push(AppConfig.pod_uri.to_s)
+    end
+
+    def subscribed_users_ids
+      object.subscribers.map(&:diaspora_handle)
```

Comment:
I just saw that `subscribers` returns the person multiple times, when you have the person in multiple aspects.

That's because [Person.in_aspects](https://github.com/diaspora/diaspora/blob/9a2798331361954c259657a96d300fbe59ad5590/app/models/person.rb#L78-L81) which is used in [User.people_in_aspects](https://github.com/diaspora/diaspora/blob/d3edbfd82962bc5cf6a995085d3b2a537d5db84f/app/models/user/querying.rb#L54) returns the person multiple times. So we should filter that somewhere instead of adding the diaspora ID multiple times to the export. (That is also broken in the federation, we encrypt the message multiple times, but send it only once, because the target-url is used as a key)